extract_from_hg.py

```diff
@@ -195,15 +196,110 @@ def commitCset(cset):
     runCommand(['git', 'commit', '-m', commitMsg, '--author=' + cset.user(),
                 '--date=' + str(csetDate)])
 
+def formatCommitMessageLine(line):
+    # Now do translations on the line.
+
+    # First, strip off any review flags from the end of line.
+    line = re.sub(r"""
+     \ *         # Match any number of spaces
+     \[*         # Match none or more of `[`, don't worry about `]` - some commit messages do `[r=smith]`
+     rs*         # Match r or rs
+     \=.*        # Match = and the rest of the line.
+    """, "", line, 0, re.VERBOSE)
+
+    # Now change any ': ' to dashes - some people use "Bug 123456: ..."
+    line = re.sub(r": ", " - ", line)
+
+    # Next change any '123456-' to add a space either side of the dash.
+    # Some people do 'Bug 123456-...'.
+    line = re.sub(r"\([0-9]]\)-", r"\1 - ", line)
+
+    # Now replace any commas at the end of the line with dots.
+    line = re.sub(r",$", ".", line)
+
+    # Finally insert '- ' at start of line for changelog formatting.
+    return "- " + line
+
+def writeChangeLog(gitRepo, headGitCommit):
+    inFile = open(CHANGELOG_FILE, "r")
+    oldLines = inFile.readlines()
+    inFile.close()
+
+    outFile = open(CHANGELOG_FILE, "w")
+    foundTBD = False
+    foundFirstDashes = False
+    outBuffer = []
+
+    # Find where to insert the new lines. We hunt down to the first TBD followed
+    # by "---" and then look for the next one. Note that 'i' gets used in the for
```

Comment:
It'd be nice to document somewhere how these TBDs might get there in the first place, and what the rationale for using them is.  Ideally this documentation would be somewhere in the release checklist, and perhaps a comment here linking to there.